vector/test.c

```diff
@@ -0,0 +1,81 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+#include "array.c"
+#include "test.h"
+
+void test_vector_create(){
+  array* arr =  vector_create();
+  ASSERT(arr != NULL);
+  ASSERT(arr->capacity == init_capacity);
+  ASSERT(arr->size == 0);
+
+  vector_free(arr);
+}
+
+void test_vector_append(){
+  array* arr =  vector_create();
+  vector_append(arr, 5); 
+  ASSERT(arr->array[0] == 5);
+  ASSERT(arr->size == 1);
+  
+  vector_free(arr);
+}
+
+void test_vector_pop(){
+  array* arr =  vector_create();
+  vector_append(arr, 5);
+  vector_append(arr, 4);
+  vector_append(arr, 3);
+  data_t popped_el = vector_pop(arr);
+  ASSERT(popped_el == 3);
+
+  vector_free(arr);
+}
+
+void test_vector_create_size(){
+  array* arr =  vector_create_size(36);
+  ASSERT(arr->capacity == 36);
+}
```

Comment:
Also, add another testcase for this function, that will check for `NULL` return.

How could we do that? Well... that's not so straightforward, but just pass a very big number, that would be bigger then someone's RAM + SWAP. E.g. the maximum possible number for `size_t` type ([check here](http://stackoverflow.com/questions/3472311/what-is-a-portable-method-to-find-the-maximum-value-of-size-t)).